node/migrations/V1589211379__create_append_only_tables.sql

```diff
@@ -0,0 +1,49 @@
+CREATE TABLE token_state_append_only (
+    id uuid PRIMARY KEY DEFAULT gen_random_uuid() NOT NULL,
+    token_id uuid NOT NULL references tokens(id),
+    state_data_json JSONB NOT NULL DEFAULT '{}',
+    created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT now()
+);
+CREATE INDEX index_token_state_append_only_uuid ON token_state_append_only (id);
+CREATE INDEX index_token_state_append_only_token_id_created_at ON token_state_append_only (token_id, created_at);
+
+CREATE OR REPLACE VIEW tokens_view AS
+SELECT
+    t.*,
+    COALESCE(tsao.state_data_json, t.initial_data_json) as additional_data_json
+FROM
+  tokens t
+LEFT JOIN
+(
+    SELECT DISTINCT ON(token_id) *
+    FROM token_state_append_only
+    ORDER BY token_id, created_at DESC
+) tsao
+ON
+    t.id = tsao.token_id;
+
+CREATE TABLE asset_state_append_only (
+    id uuid PRIMARY KEY DEFAULT gen_random_uuid() NOT NULL,
+    asset_state_id uuid NOT NULL references asset_states(id),
+    state_data_json JSONB NOT NULL DEFAULT '{}',
```

Comment:
`asset_id` is going to change too as it is hash of metadata, also probably `status`?

Reply:
So we agreed that initial_state is the state from the beginning of times, not from the checkpoint, hence asset_id won't change.